### ISSUE TITLE
crash / linux arm64

### ISSUE THREAD

**Issue by xuewei1985** [wailsapp/wails#2657]
### Description

Crash on arm64
[crash.log](https://github.com/wailsapp/wails/files/11447514/crash.log)

### To Reproduce

My logic is that the frontend page calls the backend interface every 2 seconds to refresh data, and there will be irregular crashes during this process.

The backend logic is the HTTP download function, updating the current download progress.

### Expected behaviour

Not crash

### System Details

```shell
Wails CLI v2.4.1

Scanning system - Please wait (this may take a long time)...Done.

# System

OS           | Kylin   
Version      | v10     
ID           | kylin   
Go Version   | go1.20.3
Platform     | linux   
Architecture | arm64   

# Wails

Version         | v2.4.1
Package Manager | apt   

# Dependencies

Dependency | Package Name          | Status    | Version                
*docker    | docker.io             | Installed | 20.10.7-0kylin5~20.04.2
gcc        | build-essential       | Installed | 9.3.0                  
libgtk-3   | libgtk-3-dev          | Installed | 3.24.23-1kylin2k14     
libwebkit  | libwebkit2gtk-4.0-dev | Installed | 2.28.1-1kylin1k7       
npm        | npm                   | Installed | 9.5.1                  
*nsis      | nsis                  | Available | 3.05-2                 
pkg-config | pkg-config            | Installed | 0.29.1                 
* - Optional Dependency

# Diagnosis

Your system is ready for Wails development!
Optional package(s) installation details: 
  - nsis: sudo apt install nsis

 ♥   If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

**Comment by xuewei1985** [comment]
Sometimes it crashes just after startup, such as this log
```
runtime: g 10: unexpected return pc for github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend.func1 called from 0x4000439ff0
stack: frame={sp:0x40003a6fb0, fp:0x40003a6fd0} stack=[0x40003a6000,0x40003a7000)
0x00000040003a6eb0:  0x0000000000412ed4 <runtime.chanrecv2+0x0000000000000014>  0x0000004000439ef8 
0x00000040003a6ec0:  0x00000000008713f0 <github.com/wailsapp/wails/v2/internal/frontend/desktop/linux._Cfunc_g_object_unref+0x0000000000000030>  0x0000000000942bb8 
0x00000040003a6ed0:  0x0000004000439f38  0x0000004000000000 
0x00000040003a6ee0:  0x0000000000875344 <github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).startRequestProcessor.func1+0x0000000000000034>  0x0000000000000000 
0x00000040003a6ef0:  0x0000004000130c00  0x0000004000382240 
0x00000040003a6f00:  0x0000004000308838  0x0000004000331520 
0x00000040003a6f10:  0x00000040003315a8  0x0000004000382290 
0x00000040003a6f20:  0x0000004000331668  0x0000004000044768 
0x00000040003a6f30:  0x00000000008752d0 <github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).startRequestProcessor+0x0000000000000050>  0x0000004000044768 
0x00000040003a6f40:  0x00000000008752e0 <github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).startRequestProcessor+0x0000000000000060>  0x00000040003087e0 
0x00000040003a6f50:  0x0000004000439fa0  0x000000400041c001 
0x00000040003a6f60:  0x00000000281f99c0  0x00000040000447a8 
0x00000040003a6f70:  0x0000000000873f78 <github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend.func1+0x0000000000000028>  0x00000040003097a0 
0x00000040003a6f80:  0x0000004000439fb0  0x0000007faa933958 
0x00000040003a6f90:  0x0000007faa9d0000  0x0000000027fb5fb0 
0x00000040003a6fa0:  0x00000040003a6fe0  0x0000000000000001 
0x00000040003a6fb0: <0x0000004000439ff0  0x0000007faa8e6070 
0x00000040003a6fc0:  0x0000000000000011  0x0000007f527fa560 
0x00000040003a6fd0: >0x000000000000001d  0x0000000000000000 
0x00000040003a6fe0:  0x0000000000000001  0x1fac0aa3cb01b000 
0x00000040003a6ff0:  0x000000400043b270  0x0000007fafbba5b0 
fatal error: unknown caller pc

runtime stack:
runtime.throw({0xa61ba3?, 0xff0b00?})
        /usr/local/go/src/runtime/panic.go:1047 +0x40 fp=0x7f537fac60 sp=0x7f537fac30 pc=0x440aa0
runtime.gentraceback(0x4000331520?, 0x170?, 0x7f537fb068?, 0x4000331520, 0x0, 0x0, 0x7fffffff, 0xbf1580, 0x8752e0?, 0x0)
        /usr/local/go/src/runtime/traceback.go:270 +0x14bc fp=0x7f537fafd0 sp=0x7f537fac60 pc=0x464d5c
runtime.copystack(0x4000331520, 0x1000)
        /usr/local/go/src/runtime/stack.go:934 +0x2fc fp=0x7f537fb190 sp=0x7f537fafd0 pc=0x459b1c
runtime.shrinkstack(0x4000331520)
        /usr/local/go/src/runtime/stack.go:1216 +0x124 fp=0x7f537fb1b0 sp=0x7f537fb190 pc=0x45a6c4
runtime.scanstack(0x4000331520, 0x4000039738)
        /usr/local/go/src/runtime/mgcmark.go:775 +0xf8 fp=0x7f537fb3c0 sp=0x7f537fb1b0 pc=0x42a358
runtime.markroot.func1()
        /usr/local/go/src/runtime/mgcmark.go:239 +0xb4 fp=0x7f537fb410 sp=0x7f537fb3c0 pc=0x429224
runtime.markroot(0x4000039738, 0x1e, 0x1)
        /usr/local/go/src/runtime/mgcmark.go:213 +0x1cc fp=0x7f537fb4b0 sp=0x7f537fb410 pc=0x428efc
runtime.gcDrain(0x4000039738, 0x7)
        /usr/local/go/src/runtime/mgcmark.go:1069 +0x3fc fp=0x7f537fb510 sp=0x7f537fb4b0 pc=0x42aeec
runtime.gcBgMarkWorker.func2()
        /usr/local/go/src/runtime/mgc.go:1367 +0x74 fp=0x7f537fb560 sp=0x7f537fb510 pc=0x427154
runtime.systemstack()
        /usr/local/go/src/runtime/asm_arm64.s:243 +0x6c fp=0x7f537fb570 sp=0x7f537fb560 pc=0x46f83c

goroutine 53 [GC worker (active)]:
runtime.systemstack_switch()
        /usr/local/go/src/runtime/asm_arm64.s:200 +0x8 fp=0x400004af40 sp=0x400004af30 pc=0x46f7b8
runtime.gcBgMarkWorker()
        /usr/local/go/src/runtime/mgc.go:1335 +0x210 fp=0x400004afd0 sp=0x400004af40 pc=0x426e20
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x400004afd0 sp=0x400004afd0 pc=0x471d84
created by runtime.gcBgMarkStartWorkers
        /usr/local/go/src/runtime/mgc.go:1199 +0x28

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x942bc8, 0x400034dda8)
        /usr/local/go/src/runtime/cgocall.go:157 +0x50 fp=0x400034dd70 sp=0x400034dd30 pc=0x410870
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux._Cfunc_gtk_main()
        _cgo_gotypes.go:1173 +0x30 fp=0x400034dda0 sp=0x400034dd70 pc=0x871b40
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).RunMainLoop(0x40003097a0?)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:121 +0x1c fp=0x400034ddb0 sp=0x400034dda0 pc=0x8738dc
github.com/wailsapp/wails/v2/internal/app.(*App).Run(0x400038a000)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/app/app_production.go:19 +0x58 fp=0x400034ddf0 sp=0x400034ddb0 pc=0x8813f8
github.com/wailsapp/wails/v2/pkg/application.(*Application).Run(0x40003110a0)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/pkg/application/application.go:72 +0xa8 fp=0x400034de20 sp=0x400034ddf0 pc=0x8827a8
github.com/wailsapp/wails/v2.Run(0x4000189b80?)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/wails.go:14 +0xa4 fp=0x400034de50 sp=0x400034de20 pc=0x882964
main.main()
        /home/kylin/Projects/winapps-mgr/main.go:38 +0xf00 fp=0x400034df70 sp=0x400034de50 pc=0x883890
runtime.main()
        /usr/local/go/src/runtime/proc.go:250 +0x200 fp=0x400034dfd0 sp=0x400034df70 pc=0x4430e0
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x400034dfd0 sp=0x400034dfd0 pc=0x471d84

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x4000048fa0 sp=0x4000048f80 pc=0x4434f0
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:387
runtime.forcegchelper()
        /usr/local/go/src/runtime/proc.go:305 +0xb0 fp=0x4000048fd0 sp=0x4000048fa0 pc=0x443340
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x4000048fd0 sp=0x4000048fd0 pc=0x471d84
created by runtime.init.6
        /usr/local/go/src/runtime/proc.go:293 +0x24

goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x4000049760 sp=0x4000049740 pc=0x4434f0
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:387
runtime.bgsweep(0x0?)
        /usr/local/go/src/runtime/mgcsweep.go:278 +0x98 fp=0x40000497b0 sp=0x4000049760 pc=0x42fe58
runtime.gcenable.func1()
        /usr/local/go/src/runtime/mgc.go:178 +0x28 fp=0x40000497d0 sp=0x40000497b0 pc=0x424f18
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x40000497d0 sp=0x40000497d0 pc=0x471d84
created by runtime.gcenable
        /usr/local/go/src/runtime/mgc.go:178 +0x6c

goroutine 4 [GC scavenge wait]:
runtime.gopark(0x400006e000?, 0xc7b4e0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x4000049f50 sp=0x4000049f30 pc=0x4434f0
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:387
runtime.(*scavengerState).park(0x10dca60)
        /usr/local/go/src/runtime/mgcscavenge.go:400 +0x5c fp=0x4000049f80 sp=0x4000049f50 pc=0x42ddec
runtime.bgscavenge(0x0?)
        /usr/local/go/src/runtime/mgcscavenge.go:633 +0xa8 fp=0x4000049fb0 sp=0x4000049f80 pc=0x42e388
runtime.gcenable.func2()
        /usr/local/go/src/runtime/mgc.go:179 +0x28 fp=0x4000049fd0 sp=0x4000049fb0 pc=0x424eb8
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x4000049fd0 sp=0x4000049fd0 pc=0x471d84
created by runtime.gcenable
        /usr/local/go/src/runtime/mgc.go:179 +0xac

goroutine 5 [finalizer wait]:
runtime.gopark(0x1a0?, 0x10dd440?, 0xc0?, 0x36?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x4000048580 sp=0x4000048560 pc=0x4434f0
runtime.runfinq()
        /usr/local/go/src/runtime/mfinal.go:193 +0x100 fp=0x40000487d0 sp=0x4000048580 pc=0x423fe0
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x40000487d0 sp=0x40000487d0 pc=0x471d84
created by runtime.createfing
        /usr/local/go/src/runtime/mfinal.go:163 +0x80

goroutine 6 [select]:
runtime.gopark(0x400004a778?, 0x2?, 0x68?, 0xa6?, 0x400004a774?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x400004a620 sp=0x400004a600 pc=0x4434f0
runtime.selectgo(0x400004a778, 0x400004a770, 0x0?, 0x0, 0x0?, 0x1)
        /usr/local/go/src/runtime/select.go:327 +0x68c fp=0x400004a730 sp=0x400004a620 pc=0x45313c
database/sql.(*DB).connectionOpener(0x400018da00, {0xc84478, 0x4000182c30})
        /usr/local/go/src/database/sql/sql.go:1218 +0x80 fp=0x400004a7a0 sp=0x400004a730 pc=0x691710
database/sql.OpenDB.func1()
        /usr/local/go/src/database/sql/sql.go:791 +0x30 fp=0x400004a7d0 sp=0x400004a7a0 pc=0x68fa70
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x400004a7d0 sp=0x400004a7d0 pc=0x471d84
created by database/sql.OpenDB
        /usr/local/go/src/database/sql/sql.go:791 +0x170

goroutine 7 [chan receive]:
runtime.gopark(0x4000056ef8?, 0x8713f0?, 0xb8?, 0x2b?, 0x4000056f38?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x4000056eb0 sp=0x4000056e90 pc=0x4434f0
runtime.chanrecv(0x40003087e0, 0x4000056fa0, 0x1)
        /usr/local/go/src/runtime/chan.go:583 +0x468 fp=0x4000056f40 sp=0x4000056eb0 pc=0x413358
runtime.chanrecv2(0x40003097a0?, 0x281f58c0?)
        /usr/local/go/src/runtime/chan.go:447 +0x14 fp=0x4000056f70 sp=0x4000056f40 pc=0x412ed4
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).startRequestProcessor(0x0?)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:467 +0x60 fp=0x4000056fb0 sp=0x4000056f70 pc=0x8752e0
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend.func1()
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:171 +0x28 fp=0x4000056fd0 sp=0x4000056fb0 pc=0x873f78
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x4000056fd0 sp=0x4000056fd0 pc=0x471d84
created by github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:171 +0x3dc

goroutine 8 [chan receive]:
runtime.gopark(0x40003a4ef8?, 0x8713f0?, 0xb8?, 0x2b?, 0x40003a4f38?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x40003a4eb0 sp=0x40003a4e90 pc=0x4434f0
runtime.chanrecv(0x40003087e0, 0x40003a4fa0, 0x1)
        /usr/local/go/src/runtime/chan.go:583 +0x468 fp=0x40003a4f40 sp=0x40003a4eb0 pc=0x413358
runtime.chanrecv2(0x40003097a0?, 0x281f58c0?)
        /usr/local/go/src/runtime/chan.go:447 +0x14 fp=0x40003a4f70 sp=0x40003a4f40 pc=0x412ed4
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).startRequestProcessor(0x0?)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:467 +0x60 fp=0x40003a4fb0 sp=0x40003a4f70 pc=0x8752e0
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend.func1()
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:171 +0x28 fp=0x40003a4fd0 sp=0x40003a4fb0 pc=0x873f78
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x40003a4fd0 sp=0x40003a4fd0 pc=0x471d84
created by github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:171 +0x3dc

goroutine 9 [chan receive]:
runtime.gopark(0x4000095ef8?, 0x8713f0?, 0xb8?, 0x2b?, 0x4000095f38?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x40003a5eb0 sp=0x40003a5e90 pc=0x4434f0
runtime.chanrecv(0x40003087e0, 0x4000095fa0, 0x1)
        /usr/local/go/src/runtime/chan.go:583 +0x468 fp=0x40003a5f40 sp=0x40003a5eb0 pc=0x413358
runtime.chanrecv2(0x40003097a0?, 0x281f7940?)
        /usr/local/go/src/runtime/chan.go:447 +0x14 fp=0x40003a5f70 sp=0x40003a5f40 pc=0x412ed4
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).startRequestProcessor(0x0?)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:467 +0x60 fp=0x40003a5fb0 sp=0x40003a5f70 pc=0x8752e0
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend.func1()
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:171 +0x28 fp=0x40003a5fd0 sp=0x40003a5fb0 pc=0x873f78
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x40003a5fd0 sp=0x40003a5fd0 pc=0x471d84
created by github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.NewFrontend
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:171 +0x3dc

goroutine 10 [chan receive (scan)]:
runtime.gopark(0x4000439ef8?, 0x8713f0?, 0xb8?, 0x2b?, 0x4000439f38?)
        /usr/local/go/src/runtime/proc.go:381 +0xe0 fp=0x40003a6eb0 sp=0x40003a6e90 pc=0x4434f0
runtime.chanrecv(0x40003087e0, 0x4000439fa0, 0x1)
        /usr/local/go/src/runtime/chan.go:583 +0x468 fp=0x40003a6f40 sp=0x40003a6eb0 pc=0x413358
runtime.chanrecv2(0x40003097a0?, 0x4000439fb0?)
        /usr/local/go/src/runtime/chan.go:447 +0x14 fp=0x40003a6f70 sp=0x40003a6f40 pc=0x412ed4
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Frontend).startRequestProcessor(0x7faa8e6070?)
        /home/kylin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.4.1/internal/frontend/desktop/linux/frontend.go:467 +0x60 fp=0x40003a6fb0 sp=0x40003a6f70 pc=0x8752e0
fatal error: unexpected signal during runtime execution
panic during panic
[signal SIGSEGV: segmentation violation code=0x1 addr=0x118 pc=0x463e70]

runtime stack:
runtime.throw({0xa74a14?, 0x463d60?})
        /usr/local/go/src/runtime/panic.go:1047 +0x40 fp=0x7f537fa4d0 sp=0x7f537fa4a0 pc=0x440aa0
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_unix.go:821 +0x240 fp=0x7f537fa510 sp=0x7f537fa4d0 pc=0x456e00
runtime.gentraceback(0x4000331380?, 0xff0b00?, 0x7f537fa8f8?, 0x4000331520, 0x0, 0x0, 0x64, 0x0, 0x400004afd0?, 0x0)
        /usr/local/go/src/runtime/traceback.go:258 +0x5d0 fp=0x7f537fa890 sp=0x7f537fa520 pc=0x463e70
runtime.traceback1(0x4000331520?, 0x0?, 0x7f537faab8?, 0x4000331520, 0xa51d7c?)
        /usr/local/go/src/runtime/traceback.go:776 +0x154 fp=0x7f537faa50 sp=0x7f537fa890 pc=0x465964
runtime.traceback(...)
        /usr/local/go/src/runtime/traceback.go:723
runtime.tracebackothers.func1(0x4000331520)
        /usr/local/go/src/runtime/traceback.go:992 +0x104 fp=0x7f537faa90 sp=0x7f537faa50 pc=0x4667c4
runtime.forEachGRace(0x7f537fab08)
        /usr/local/go/src/runtime/proc.go:604 +0x60 fp=0x7f537faac0 sp=0x7f537faa90 pc=0x443ff0
runtime.tracebackothers(0x400036a9c0?)
        /usr/local/go/src/runtime/traceback.go:978 +0xd8 fp=0x7f537fab30 sp=0x7f537faac0 pc=0x466698
runtime.dopanic_m(0x400036a9c0, 0x4423c8?, 0x1?)
        /usr/local/go/src/runtime/panic.go:1269 +0x258 fp=0x7f537fabb0 sp=0x7f537fab30 pc=0x441318
runtime.fatalthrow.func1()
        /usr/local/go/src/runtime/panic.go:1123 +0x44 fp=0x7f537fabf0 sp=0x7f537fabb0 pc=0x440d94
runtime.fatalthrow(0x537fac40?)
        /usr/local/go/src/runtime/panic.go:1120 +0x58 fp=0x7f537fac30 sp=0x7f537fabf0 pc=0x440d38
runtime.throw({0xa61ba3?, 0xff0b00?})
        /usr/local/go/src/runtime/panic.go:1047 +0x40 fp=0x7f537fac60 sp=0x7f537fac30 pc=0x440aa0
runtime.gentraceback(0x4000331520?, 0x170?, 0x7f537fb068?, 0x4000331520, 0x0, 0x0, 0x7fffffff, 0xbf1580, 0x8752e0?, 0x0)
        /usr/local/go/src/runtime/traceback.go:270 +0x14bc fp=0x7f537fafd0 sp=0x7f537fac60 pc=0x464d5c
runtime.copystack(0x4000331520, 0x1000)
        /usr/local/go/src/runtime/stack.go:934 +0x2fc fp=0x7f537fb190 sp=0x7f537fafd0 pc=0x459b1c
runtime.shrinkstack(0x4000331520)
        /usr/local/go/src/runtime/stack.go:1216 +0x124 fp=0x7f537fb1b0 sp=0x7f537fb190 pc=0x45a6c4
runtime.scanstack(0x4000331520, 0x4000039738)
        /usr/local/go/src/runtime/mgcmark.go:775 +0xf8 fp=0x7f537fb3c0 sp=0x7f537fb1b0 pc=0x42a358
runtime.markroot.func1()
        /usr/local/go/src/runtime/mgcmark.go:239 +0xb4 fp=0x7f537fb410 sp=0x7f537fb3c0 pc=0x429224
runtime.markroot(0x4000039738, 0x1e, 0x1)
        /usr/local/go/src/runtime/mgcmark.go:213 +0x1cc fp=0x7f537fb4b0 sp=0x7f537fb410 pc=0x428efc
runtime.gcDrain(0x4000039738, 0x7)
        /usr/local/go/src/runtime/mgcmark.go:1069 +0x3fc fp=0x7f537fb510 sp=0x7f537fb4b0 pc=0x42aeec
runtime.gcBgMarkWorker.func2()
        /usr/local/go/src/runtime/mgc.go:1367 +0x74 fp=0x7f537fb560 sp=0x7f537fb510 pc=0x427154
runtime.systemstack()
        /usr/local/go/src/runtime/asm_arm64.s:243 +0x6c fp=0x7f537fb570 sp=0x7f537fb560 pc=0x46f83c
```

**Comment by xuewei1985** [comment]
I have found a phenomenon during testing that adding the -debug parameter during compilation prevents the crash. Hopefully, this could be a clue.

**Comment by stffabi** [comment]
Thanks for reporting this.

Have you tried updating your `webkit2gtk 2.28.1-1kylin1k7` version, because 2.28.1 seem pretty old and is about 3 years old. A production build uses custom schemes and custom scheme requests whereas a dev build before `2.5.0` use a normal http server for providing the assets. So it could possibly be that this webkit2gtk version had problems with custom schemes and requests from that.

**Comment by xuewei1985** [comment]
Thank you for your reply

Okay, I'll try updating the webkit2GTK version;

Also, I have updated Wails to version 2.5.0 on my end.

**Comment by leaanthony** [comment]
@xuewei1985 - any updates?

**Comment by xuewei1985** [comment]
> @xuewei1985 - any updates?

The current situation on my end is that I'm trying to compile a higher version of WebKit2GTK, but the compilation process is slow, and I haven't been able to successfully compile it yet. For now, I have compiled a version using the "-debug" parameter to proceed with my work. Once I have finished my current tasks, I will conduct upgrade testing for WebKit2GTK.

**Comment by xuewei1985** [comment]
I tried it in Ubuntu 22.04 doctor and it didn’t crash. Wails doctor is as follows:
```
Wails CLI v2.5.1

 SUCCESS  Done.                                                                                                                                                                                  

# System

OS           | Ubuntu  
Version      | 22.04   
ID           | ubuntu  
Go Version   | go1.20.4
Platform     | linux   
Architecture | arm64   

# Wails

Version         | v2.5.1
Package Manager | apt   

# Dependencies

Dependency | Package Name          | Status    | Version                  
*docker    | docker.io             | Available | 20.10.21-0ubuntu1~22.04.3
gcc        | build-essential       | Installed | 11.3.0                   
libgtk-3   | libgtk-3-dev          | Installed | 3.24.33-1ubuntu2         
libwebkit  | libwebkit2gtk-4.0-dev | Installed | 2.38.6-0ubuntu0.22.04.1  
npm        | npm                   | Installed | 9.5.1                    
*nsis      | nsis                  | Available | 3.08-2                   
pkg-config | pkg-config            | Installed | 0.29.2-1ubuntu3          
* - Optional Dependency

# Diagnosis

Your system is ready for Wails development!
Optional package(s) installation details: 
  - docker: sudo apt install docker.io
  - nsis: sudo apt install nsis

 ♥   If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

**Comment by stffabi** [comment]
I think we can close this issue, because it seemed to be related to the old WebKit2GTK version.

Please feel free to reopen if you see any further problem.
